### DISCUSSION TITLE
A document listing as conditional content

### DISCUSSION THREAD

**Discussion by andrewpbray** [quarto-dev/quarto-cli#4635] [Q&A] [answered]
Hello, I've just tried putting a document listing as conditional content and it didn't work. Here's a MWE.

[conditional-listing.zip](https://github.com/quarto-dev/quarto-cli/files/10892937/conditional-listing.zip)

The operative part is in this doc:

```
---
title: "Conditional-listing"
listing:
  id: blog
  contents:
    - post-1.qmd
    - post-2.qmd
---

This is a Quarto website.

To learn more about Quarto websites visit <https://quarto.org/docs/websites>.

:::{.content-hidden when-profile="noblog"}

:::{#blog}
:::

:::
```

I had hoped that on `quarto render` I'd see the blog but on `quarto render --profile noblog` I would not, but alas, both versions have the blog.

I'm guessing this has something to do with the order of operations on render?

Thanks for any thoughts / suggestions.

**Comment by mcanouil** [comment, marked as answer]
This won't work.
`:::{#blog}` is only a placeholder, it does not trigger the creation of a listings, it only overrides the default location of the listing.

The only way I think you might be able to do what you want is to use pre/post-render script to write the "index.qmd" file depending on profile/environment (see <https://quarto.org/docs/projects/profiles.html>).

Note that an issue to track this has be opened (can be seen on the right of your post): #4651
